### PR TITLE
PAYARA-4030 Upgrade HK2 to 2.6.0

### PR DESCRIPTION
**Pull request by Cousjava**
Includes upgrading asm to 7.1

### CODE REVIEW

File: nucleus/packager/nucleus-hk2/pom.xml [hunk 1]
```diff
@@ -97,15 +97,16 @@
...
             <groupId>org.glassfish.hk2.external</groupId>
-            <artifactId>asm-repackaged</artifactId>
+            <artifactId>aopalliance-repackaged</artifactId>
         </dependency>
         <dependency>
-            <groupId>org.glassfish.hk2.external</groupId>
-            <artifactId>aopalliance-repackaged</artifactId>
+            <groupId>fish.payara.external</groupId>
+            <artifactId>tiger-types</artifactId>
+            <version>${project.version}</version>
```
Comment (Cousjava) [inline nucleus/packager/nucleus-hk2/pom.xml]: This is not a new dependency, but previously HK2 repackaged it and included it. HK2 no longer uses it, so we have to repackage it ourselves.

**Comment by Cousjava** [comment]
Jenkins test please

File: appserver/registration/glassfish-registration/src/main/java/com/sun/enterprise/registration/glassfish/ModuleMap.java [hunk 2]
```diff
@@ -159,7 +159,7 @@
                 put("org.glassfish.hk2.config", 100);
                 put("org.glassfish.hk2.config-types", 101);
                 put("org.glassfish.hk2.core", 102);
-                put("org.glassfish.hk2.external.asm-repackaged", 103);
+                //put("org.glassfish.hk2.external.asm-repackaged", 103);
```
Comment (jbee) [inline appserver/registration/glassfish-registration/src/main/java/com/sun/enterprise/registration/glassfish/ModuleMap.java]: Shouldn't these be removed? (also the ones already there?) Or is there a particular reason to keep them?

**Comment by dmatej** [comment]
Ok, the crash is explained, and it is not related to this PR.

**Pull request merged**